lib/note/index.js

```diff
@@ -232,7 +234,100 @@ function listMyNotes (req, res) {
   }
 }
 
+const deleteNote = async (req, res) => {
+  if (req.isAuthenticated()) {
+    const noteId = await Note.parseNoteIdAsync(req.params.noteId)
+    try {
+      const destroyed = await Note.destroy({
+        where: {
+          id: noteId,
+          ownerId: req.user.id
+        }
+      })
+      if (!destroyed) {
+        logger.error('Delete note failed: Make sure the noteId and ownerId are correct.')
+        return errorNotFound(req, res)
+      }
+
+      historyDelete(req, res)
+
+      if (realtime.isNoteExistsInPool(noteId)) {
+        const note = realtime.getNoteFromNotePool(noteId)
+        realtime.disconnectSocketOnNote(note)
+      }
+
+      res.send({
+        status: 'ok'
+      })
+    } catch (err) {
+      logger.error('Delete note failed: Internal Error.')
+      return errorInternalError(req, res)
+    }
+  } else {
+    return errorForbidden(req, res)
+  }
+}
+
+const updateNote = async (req, res) => {
+  if (req.isAuthenticated()) {
+    const noteId = await Note.parseNoteIdAsync(req.params.noteId)
+    try {
+      const note = await Note.findOne({
+        where: {
+          id: noteId
+        }
+      })
+      if (!note) {
+        logger.error('Update note failed: Can\'t find the note.')
+        return errorNotFound(req, res)
+      }
+
+      if (realtime.isNoteExistsInPool(noteId)) {
+        logger.error('Update note failed: There are online users opening this note.')
+        return res.status('403').send({ status: 'error', message: 'Update API can only be used when no users is online' })
+      }
+
+      const now = Date.now()
+      const content = req.body.content
+      const updated = await note.update({
+        content: content,
+        lastchangeAt: moment(now).format('YYYY-MM-DD HH:mm:ss'),
```

Comment:
Update lastchangeAt with date directly is okay:

```suggestion
        lastchangeAt: now,
```

Reply:
Fix.